internal/service/dynamodb/table_replica.go

```diff
@@ -161,7 +161,16 @@ func resourceTableReplicaCreate(ctx context.Context, d *schema.ResourceData, met
 	if tfresource.TimedOut(err) {
 		_, err = conn.UpdateTable(ctx, input, optFn)
 	}
-
+	if tfawserr.ErrMessageContains(err, errCodeValidationException, "read capacity should") || tfawserr.ErrMessageContains(err, errCodeValidationException, "write capacity should") {
+		// We cannot tell if there is a pending auto scaling rule, so we cannot simply retry or else a table with billing_mode = "PROVISIONED" will also retry until the timeout.
+		return append(diags, errs.NewErrorDiagnostic(
+			"Incompatible Source Table Configuration",
+			"For a DynamoDB Table Replica, the source table must either have a billing mode of \"PAY_PER_REQUEST\" or the read and write capacities must be auto scaled. "+
+				"If you expect the read and write capacities to be auto scaled, make sure that the auto scaling policies are applied to the source table before creating the replica. "+
+				"To do this, add a \"depends_on\" argument which references the auto scaling policies.\n\n"+
```

Comment:
Maybe worth moving this into a `const` since it's in two places, just so the content doesn't inadvertently diverge?